rspace/src/main/tut/Tutorial.md

```diff
@@ -408,9 +408,69 @@ val pres14 = space.produce(Channel("friends"), erin, persist = true)
 println(space.store.toMap)
 ```
 
+### History & rollback
+
+It is possible to save the current state of RSpace in the form of a checkpoint (root hash of a given Merkle Patricia Trie).
+```scala
+val checkpoint = space.createCheckpoint()
+val checkpointHash = checkpoint.root
+```
+
+To rollback the state of the RSpace to a given checkpoint one simply calls the `reset` method with the hash of the root of the checkpoint provided as parameter.
+```scala
+space.reset(checkpointHash)
+```
+
+Let's see how this works in practice. We'll start by creating a new, untouched RSpace followed by a consume operation which should put data and a continuation at given channel.
+```tut
+    val storePath2: Path = Files.createTempDirectory("rspace-address-book-example-")
```

Comment:
Don't indent like this.  It makes the tut-generated REPL examples look pretty weird.